src/core/adt/show.js

```diff
@@ -0,0 +1,26 @@
+const { tagSymbol, typeSymbol } = require('./core');
+
+const displayValues = (object) => Object.getOwnPropertyNames(object)
```

Comment:
We should use a `key: value` pair serialisation here, similar to what JSON.stringify does. This avoids relying on key order (which is only guaranteed to be insertion order in ES2015), and helps people with what they should be destructuring in the `.cata()` method. I think `Object.keys` makes more sense, we should clarify somewhere that the ADT module and its derivations only consider enumerable keys. Something like:

```js
const displayValues = (object) =>
  Object.keys(object)
        .map((key) => [`${JSON.stringify(key)}: ${JSON.stringify(String(value))}`])
        .join(', ')
```